### PR TITLE
Feature/cache + others

### PR DESCRIPTION
**Pull request by garyluu**
Initially added the npm cache as per Cypress instruction, somehow turned into random fixes for other issues

- Fixes the collections and collection store merge.
- Added NPM cache (even keeping the typo), doesn't seem to have any effects.  Can revert if causing issues.
- Wait for the webservice in hopes that it might fix the psql timeout issue
- Use activatedRoute for getting the path parameters
- Many DAG changes, fixes the sidebar overlapping (ga4gh/dockstore#1993), barely  increased performance by rendering the DAG outside of Angular (to prevent change detection).  Can't seem to test this in Cypress because full screen functionality must be user driven.
- Reduce Date function calls by a bajllion (should get rid of that function altogether eventually).  Format changed slightly (Mar. 29 2018 at 6:56:41 p.m. to Mar 29, 2018, 6:56:41 PM)

### CODE REVIEW

File: src/app/workflow/dag/dag.component.ts [hunk 1]
```diff
@@ -28,18 +27,31 @@ import { DagQuery } from './state/dag.query';
 import { DagService } from './state/dag.service';
 import { DagStore } from './state/dag.store';
 
+/**
+ * This is the DAG tab
+ * TODO: Not have a fixed 500px normal sized DAG in case people are using different height screens
+ * TODO: Material tooltips to appear in fullscreen mode.
+ * The matTooltip DOM is in a seperate div than the fullscreen element's div, that's why it doesn't show
+ * TODO: Performance improvements
```
Comment (agduncan94) [inline src/app/workflow/dag/dag.component.ts]: Should make issues for the TODOs

Comment (garyluu) [reply src/app/workflow/dag/dag.component.ts]: ur alive!

File: .travis.yml [hunk 2]
```diff
@@ -7,6 +7,11 @@ services:
   - postgresql
 jdk:
   - oraclejdk8
+cache:
+  npm: true # this cashes ~/.npm folder on Travis
```
Comment (denis-yuen) [inline .travis.yml]: caches

File: cypress/integration/group2/workflowDetails.ts [hunk 3]
```diff
@@ -114,14 +114,18 @@ describe('Dockstore Workflow Details', () => {
...
       cy
         .get('#dag_fullscreen')
-        .click()
-        .get('#dag-col')
-        .should('have.class', 'fullscreen-element')
-        .get('#dag_fullscreen')
-        .click()
-        .should('not.have.class', 'fullscreen-element');
+        .click();
+        // Cypress or electron can't initiate fullscreen because:
```
Comment (denis-yuen) [inline cypress/integration/group2/workflowDetails.ts]: comment on when this would get reactivated if you want to keep this commented code

**Pull request merged**